### PR TITLE
Enhancement/detector last used

### PR DESCRIPTION
**Pull request by karanshah03**
Upgraded detector path from v2 to v3 and updated detector manager to update detector's `lastAccessedTime` every time it is invoked to keep track of detector usage.

### CODE REVIEW

File: anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java [hunk 1]
```diff
@@ -261,7 +266,28 @@ private void checkMappedMetricData(MappedMetricData mappedMetricData) {
...
+        if (updateDurationInSeconds <= 0 || detectorsLastUsedTimeToBeUpdated.isEmpty()) {
+            return;
+        }
+
+        int queueSize = detectorsLastUsedTimeToBeUpdated.size();
+        //Update last used time only for 1/4th of queue's size at one go for better performance
+        for (int i = 0; i < queueSize / 4; i++) {
+            UUID uuid = detectorsLastUsedTimeToBeUpdated.remove();
+            if (uuid != null) {
+                detectorSource.updatedDetectorLastUsed(uuid);
```
Comment (hridyeshpant) [inline anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java:288]: but still, we are making a single call per UUID? 
Can we use bulk API call and construct one single request for 1/4 detectors at a time?

Comment (karanshah03) [reply anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java:288]: yeah. I already added a comment to explore on that option. we can do this in second pass.

Comment (karanshah03) [reply anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java:288]: also, update calls are made via scheduler now and not for every metric that comes in.

Comment (karanshah03) [reply anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/DetectorManager.java:288]: also, update calls are made via scheduler now and not for every metric that comes in.

**Pull request merged**